### ISSUE TITLE
[Bug] All grpc thread hang due to the disk corruption

### ISSUE THREAD

**Issue by zuston** [apache/incubator-uniffle#1956]
### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### Search before asking

- [X] I have searched in the [issues](https://github.com/apache/incubator-uniffle/issues?q=is%3Aissue) and found no similar issues.

### Describe the bug

In our production environment, there was a shuffle-server hang, that is blocked on the LocalFileDeleteHandler which held the app's write lock and then all grpc threads were blocked.

And the root cause is the disk broken.

![image](https://github.com/user-attachments/assets/7fab7834-6804-4db5-8038-1dade6058575)

### Affects Version(s)

master

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

**Comment by zhengchenyu** [comment]
@zuston Can you please add a new PR to branch-0.9? I think it is a critical fix. But can not cherry-pick to branch-0.9.

**Comment by zhengchenyu** [comment]
app lock is introduced by #1593, not merged to branch-0.9, remove from 0.9.1 milestone.
